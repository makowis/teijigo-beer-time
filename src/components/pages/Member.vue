<template>
  <tbt-default>
    <div class="member">
      <tbt-h1>MEMBER</tbt-h1>
      <table class="member-table">
        <thead>
          <tr>
            <th>名前</th>
            <th>担当</th>
            <th>自己紹介</th>
          </tr>
        </thead>
        <tbody>
          <tr v-for="member in members" v-bind:key="member.name">
            <td>{{ member.name }}</td>
            <td v-html="member.charge"></td>
            <td v-html="member.profile"></td>
          </tr>
        </tbody>
      </table>
    </div>
  </tbt-default>
</template>

<script lang="ts">
import Vue from 'vue';
import Component from 'vue-class-component';
import members from '@/resources/members';
import TbtDefault from '@/components/templates/TbtDefault.vue';
import TbtH1 from '@/components/atoms/TbtH1.vue';

@Component({
  components: {
    TbtDefault,
    TbtH1,
  },
})
export default class Member extends Vue {
  members = members;
}
</script>

<style scoped>
.member-table {
  max-width: 90%;
  margin: auto;
  border: solid 1px black;
  border-collapse: separate;
}

.member-table td,
.member-table th {
  border: solid 1px black;
  padding: 5px 10px;
}
</style>
