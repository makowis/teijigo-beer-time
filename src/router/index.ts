import Vue from 'vue';
import Router from 'vue-router';
import Top from '@/components/pages/Top.vue';
import Disco from '@/components/pages/Disco.vue';
import Bbs from '@/components/pages/Bbs.vue';
import Member from '@/components/pages/Member.vue';
import Beenya from '@/components/pages/Beenya.vue';
import NotFound from '@/components/pages/NotFound.vue';

Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'Top',
      component: Top,
    },
    {
      path: '/disco',
      name: 'Disco',
      component: Disco,
    },
    {
      path: '/bbs',
      name: 'Bbs',
      component: Bbs,
    },
    {
      path: '/member',
      name: 'Member',
      component: Member,
    },
    {
      path: '/beenya',
      name: 'Beenya',
      component: Beenya,
    },
    {
      path: '*',
      name: 'NotFound',
      component: NotFound,
    },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (savedPosition) {
      return savedPosition;
    }
    return { x: 0, y: 0 };
  },
});
