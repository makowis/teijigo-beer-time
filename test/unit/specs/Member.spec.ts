import Vue from 'vue';
import Member from '@/components/pages/Member.vue';
import router from '@/router/index';

describe('Member.vue', () => {
  const Constructor = Vue.extend(Member);
  const vm = new Constructor({ router }).$mount();

  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot();
  });
});
