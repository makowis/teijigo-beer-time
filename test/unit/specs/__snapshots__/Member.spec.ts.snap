// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member.vue should match the snapshot 1`] = `
<div
  id="app"
>
  <nav
    class="menuContainer"
    id="menu-container"
  >
    <ul
      class="menu"
    >
      <li
        class="menuItem"
      >
        <a
          class="router-link-exact-active router-link-active"
          href="/"
        >
           トップ 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/disco"
        >
           DISCOGRAPHY 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/member"
        >
           メンバー紹介 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/bbs"
        >
           一言掲示板 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          class=""
          href="/beenya"
        >
           びーにゃの部屋 
        </a>
      </li>
       
      <li
        class="menuItem"
      >
        <a
          href="https://teijigo.booth.pm/items/1093403"
          target="_blank"
        >
          
        通販(BOOTH)
      
        </a>
      </li>
    </ul>
  </nav>
   
  <div
    class="mainContainer"
    id="main-container"
  >
    <a
      class="router-link-exact-active router-link-active"
      href="/"
    >
      <img
        class="logo"
        src="@/assets/logo_1.png"
      />
    </a>
     
    <div
      class="member"
    >
      <h1
        class="h1"
      >
        MEMBER
      </h1>
       
      <table
        class="member-table"
      >
        <thead>
          <tr>
            <th>
              名前
            </th>
             
            <th>
              担当
            </th>
             
            <th>
              自己紹介
            </th>
          </tr>
        </thead>
         
        <tbody>
          <tr>
            <td>
              さかもとP
            </td>
             
            <td>
              なばかりプロデューサー兼テイジゴビールタイマー
            </td>
             
            <td>
              ビール大好き！！！！！！
            </td>
          </tr>
          <tr>
            <td>
              ビール飲まない社員K
            </td>
             
            <td>
              人の褌や自分の褌で相撲を取ったりする

              <br />
              
（主にトラックメーカー）
            </td>
             
            <td>
              ビールじゃなくてハイボールなら飲みます。

              <br />
              
定時後ハイボールタイム、よろしこ
            </td>
          </tr>
          <tr>
            <td>
              yokizo
            </td>
             
            <td>
              えらそーにする
            </td>
             
            <td>
              リハビリ中で戦力になれないので、えらそーにしてます。
            </td>
          </tr>
          <tr>
            <td>
              WestTail
            </td>
             
            <td>
              ProTools見習い
            </td>
             
            <td>
              周囲の素晴らしい才能をリスペクトしつつ頑張りたいと思います
            </td>
          </tr>
          <tr>
            <td>
              ケロ
            </td>
             
            <td>
              ミュージさん
            </td>
             
            <td>
              ウィットに富んだフレキシブルな音楽でナウなヤングにバカウケ
            </td>
          </tr>
          <tr>
            <td>
              Gutch
            </td>
             
            <td>
              曲に効果音を突っ込む
            </td>
             
            <td>
              ビール飲めない！！！！！
            </td>
          </tr>
          <tr>
            <td>
              みーやん
            </td>
             
            <td>
              ハモるの大好きはもりびと
            </td>
             
            <td>
              ボーカルじゃなくてコーラスが得意
            </td>
          </tr>
          <tr>
            <td>
              台湾豆乳を飲む人
            </td>
             
            <td>
              作業に参加しない係
            </td>
             
            <td>
              ﾆﾎﾝｺﾞワカリマセン
            </td>
          </tr>
          <tr>
            <td>
              マッキー
            </td>
             
            <td>
              BPM200以上
            </td>
             
            <td>
              お酒大好き！！！！！！
            </td>
          </tr>
          <tr>
            <td>
              Ag+
            </td>
             
            <td>
              テレアポ
            </td>
             
            <td>
              どこでもアポとります！
            </td>
          </tr>
          <tr>
            <td>
              管理人MAKO
            </td>
             
            <td>
              サイトの管理とお茶汲み
            </td>
             
            <td>
              サイトの管理人です。

              <br />
              
音楽作れないのでホムペ担当になりました。

              <br />
              
昔懐かしい雰囲気のサイトを作って行きたいと思います！
            </td>
          </tr>
        </tbody>
      </table>
    </div>
     
    <footer
      class="footer"
    >
      <p>
        ©定時後ビールタイム
      </p>
    </footer>
  </div>
</div>
`;
